### PR TITLE
Add extra endpoints to the aleo node

### PR DESCRIPTION
**Pull request by jrchatruc**
## Motivation

We need extra endpoints for integration and testing

New endpoints:

- `get_transactions` (Lists transactions for a given block by height)
- `get_transaction_by_hash` (Retrieves a transaction by hash/transaction id)

## Related PRs

This depends on this PR being merged beforehand
https://github.com/AleoHQ/snarkVM/pull/996

### CODE REVIEW

File: cli/helpers/server.rs [hunk 1]
```diff
@@ -226,6 +240,18 @@ impl<N: Network> Server<N> {
         Ok(reply::json(&ledger.ledger.read().get_block(height).or_reject()?))
     }
 
+    /// Returns the transactions for the given block height.
+    async fn get_transactions(height: u32, ledger: Arc<Ledger<N>>) -> Result<impl Reply, Rejection> {
+        Ok(reply::json(&ledger.ledger.read().get_transactions(height).or_reject()?))
+    }
+
+    /// Returns the transactions for the given block height.
+    async fn get_transaction_by_hash(hash: String, ledger: Arc<Ledger<N>>) -> Result<impl Reply, Rejection> {
```
Comment (raychu86) [inline cli/helpers/server.rs]:
```suggestion
    async fn get_transaction(transaction_id: N::TransactionID, ledger: Arc<Ledger<N>>) -> Result<impl Reply, Rejection> {
```

File: cli/helpers/server.rs [hunk 2]
```diff
@@ -226,6 +240,18 @@ impl<N: Network> Server<N> {
...
+        Ok(reply::json(
+            &ledger.ledger.read().get_transaction_by_hash(&hash).or_reject()?,
```
Comment (raychu86) [inline cli/helpers/server.rs]:
```suggestion
            &ledger.ledger.read().get_transaction(transaction_id).or_reject()?,
```

File: cli/helpers/server.rs [hunk 3]
```diff
@@ -146,6 +146,18 @@ impl<N: Network> Server<N> {
...
+        let get_transactions = warp::get()
+            .and(warp::path!("testnet3" / "transactions" / u32))
+            .and(with(ledger.clone()))
+            .and_then(Self::get_transactions);
+
+        // GET /testnet3/transaction/{hash}
+        let get_transaction_by_hash = warp::get()
+            .and(warp::path!("testnet3" / "transaction" / String))
+            .and(with(ledger.clone()))
+            .and_then(Self::get_transaction_by_hash);
```
Comment (raychu86) [inline cli/helpers/server.rs]: This section will need to be updated to `get_transaction` convention as well

File: Cargo.toml [hunk 4]
```diff
@@ -64,7 +64,7 @@ version = "1"
 [dependencies.snarkvm]
 #path = "../snarkVM"
 git = "https://github.com/AleoHQ/snarkVM.git"
-rev = "f7f5b13"
+branch = "extend-ledger-api"
```
Comment (raychu86) [inline Cargo.toml]: Let's stick with `rev`. You can use `rev` from branches.

Note in the `./rust` folder there is a `Cargo.toml` snarkVM `rev` that also needs to be updated

**Comment by facundoolano** [comment]
Addressed your suggestions @raychu86

**Review by raychu86 — APPROVED** [review]

**Pull request merged**